aya/src/programs/mod.rs

```diff
@@ -971,3 +972,81 @@ impl ProgramInfo {
         Ok(ProgramInfo(info))
     }
 }
+
+/// Programs is an Iterator over loaded eBPF programs.
+pub struct Programs {
+    current: u32,
+    done: bool,
```

Comment:
I think this should be called `error` instead. You're not setting it when the
syscall returns ENOENT (and there's no need to)